pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -111,14 +112,8 @@ async def agent_model(
     ) -> AgentModel:
         check_allow_model_requests()
         tools = [self._map_tool_definition(r) for r in function_tools]
-        if result_tools:
-            tools += [self._map_tool_definition(r) for r in result_tools]
-        return OpenAIAgentModel(
-            self.client,
-            self.model_name,
-            allow_text_result,
-            tools,
-        )
+        response_format = self._map_response_format(result_tools[0]) if result_tools else None
```

Comment:
If there are multiple result tools (which I believe is the case when the result type is a union), we would definitely need to include all the options in the allowed output

Reply:
Also
```suggestion
        response_format = self._map_response_format(result_tools[0]) if result_tools and not allow_text_result else None
```
(Or, find a way for the response format to allow raw strings)

Reply:
Regarding multiple result tools: I couldn't find a case where the number of result tools is more than 1. Even in the case of a union. If you have any examples, where we have more than one result tools, please tell me.

Regarding the `allow_text_results` parameter: Please checkout @samuelcolvin's comment. The response format should allow raw strings, if we set `result_type=str`.